core/src/main/java/org/mule/runtime/core/internal/exception/OnErrorPropagateHandler.java

```diff
@@ -34,13 +40,22 @@ public boolean acceptsAll() {
   protected Function<CoreEvent, CoreEvent> beforeRouting(Exception exception) {
     return event -> {
       event = super.beforeRouting(exception).apply(event);
-      if (!isRedeliveryExhausted(exception)) {
+      if (!isRedeliveryExhausted(exception) && isOwnedTransaction()) {
         rollback(exception);
       }
       return event;
     };
   }
 
+  private boolean isOwnedTransaction() {
+    TransactionAdapter transaction = (TransactionAdapter) TransactionCoordination.getInstance().getTransaction();
+    if (transaction == null || !transaction.getComponentLocation().isPresent()) {
+      return false;
+    }
+    return transaction.getComponentLocation().get().getRootContainerName()
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Call "Optional#isPresent()" before accessing the value. ([squid:S3655](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3655))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7640&issues=AWirRZ1DotIXnmsUDiqO&open=AWirRZ1DotIXnmsUDiqO)

[ISSUE_KEY]: AWirRZ1DotIXnmsUDiqO